ERCS/erc-7682.md

```diff
@@ -72,10 +72,104 @@ type AuxiliaryFundsCapability = {
+          "amount": "0x0de0b6b3a7640000" // 1 DAI
+        }
+      }
+    }
+  }
+}
+```
+
+The wallet, upon receiving this request, can:
+- Bridge or swap the required DAI from Chain Y to Chain X
+- Ensure the user has enough native asset (e.g., ETH) for gas on Chain X
+- Complete the deposit call to Aave on Chain X
+
+This mechanism allows advanced wallets to abstract away the complexity of cross-chain or offchain funding, enabling seamless user experiences even when the user's onchain balance is insufficient on the target chain.
+
+### Auxiliary Funds and Atomic Execution
+
+Auxiliary funds provisioning (bridging, swapping, or other asset retrieval operations) is independent of the `wallet_sendCalls` execution lifecycle. The `atomicRequired` field applies only to the call bundle execution, not to auxiliary funds provisioning.
+
+### Error Codes
+
+The following error codes are defined for auxiliary funds provisioning failures.
+
+| Code     | Message                           | Description |
+|----------|-----------------------------------|-------------|
+| -32080   | Auxiliary funds provisioning failed | Wallet attempted to provision funds but failed (e.g., bridge/swap error). |
+| -32081   | Asset not supported             | The requested asset is not available through the wallet’s auxiliary fund system. |
+| -32082   | Auxiliary funds timeout         | The provisioning operation took too long or timed out. |
+| -32083   | Auxiliary funds capability not available | The wallet no longer supports auxiliary funds on the requested chain. |
+| -32084   | Invalid assetsRequiredForExecution | The structure of the assetsRequiredForExecution object is malformed. |
+
+Applications SHOULD use the `wallet_getCallsStatus` method to track the status of call bundles that involve auxiliary funds provisioning, as the provisioning process may take time to complete. The status response will indicate whether the auxiliary funds provisioning is in progress, completed, or failed.
+
+### App Implementation Guidance
 
 When an app sees that a connected wallet has access to auxiliary funds via the `auxiliaryFunds` capability in a `wallet_getCapabilities` response, the app SHOULD NOT block users from taking actions on the basis of asset balance checks.
 
+Apps MAY include the `assetsRequiredForExecution` parameter in the `capabilities.auxiliaryFunds` object to ensure the wallet has the necessary information to provision assets as needed for successful execution.
+
+### Amount Validation
+
+The wallet MUST cross-check the `amount` value in `assetsRequiredForExecution` against the `decimal()` value specified on the `asset`'s contract to ensure that the amount is interpreted correctly according to the asset's decimal precision.
+
+### Gas Estimation and Native Asset Handling
+- Applications SHOULD include an estimate for the required native asset (e.g., ETH for gas) in `assetsRequiredForExecution` when possible.
```

Comment:
is this necessary if we've already determined the wallet capable of estimating the gas fee correctly without dapp input?

Reply:
oh whoops I meant to remove this!

Reply:
https://github.com/ethereum/ERCs/pull/1126/commits/dd3a05e0af065da03d7ce5a2819dce45c53d429f